base/compiler/abstractinterpretation.jl

```diff
@@ -49,6 +57,7 @@ function abstract_call_gf_by_type(interp::AbstractInterpreter, @nospecialize(f),
         if !isa(tname, DataType)
             # can't track the backedge to the ctor right now
             # for things like Union
+            add_remark!(interp, sv, "The called constructor is too complicated")
```

Comment:
```suggestion
            add_remark!(interp, sv, "Deciding to skip computing the result due to arbitrary reasons")
```

I have no clue what this `ftname === _TYPE_NAME` condition means (from 0b8f0b7dc40), but looks really arbitrarily random these days to return from here.

Reply:
Are you saying we should just remove this whole code path? The suggested message seems like it could cause frustration for people seeing it.

Reply:
Yeah, I couldn't tell you why these specific conditions are so bad. "Too complicated" seemed like it was trying to be too PC and cause the same frustrations.

Reply:
I'll put up a PR to remove it and see what breaks :)